src/test/files/AvoidNPlusOneProblemInJPAEntitiesCheckIssue.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * creedengo - Java language - Provides rules to reduce the environmental footprint of your Java programs
+ * Copyright © 2024 Green Code Initiative (https://green-code-initiative.org/)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.greencodeinitiative.creedengo.java.checks;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.*;
+
+public class AvoidNPlusOneProblemInJPAEntitiesCheckIssue {
+
+    @Autowired
+    private AuthorRepository authorRepository;
+
+    public List<Author> smellGetAllAuthors() {
+        List<Author> authors = authorRepository.findAll();
+        for (Author author : authors) {
+            List<Book> books = author.getBooks(); // Noncompliant {{ Detection of the "N+1 problem" on Spring Data JPA repositories }}
+        }
+        return authors;
+    }
+
+
+    public class Author {
```

Comment:
I don't see JPA annotations to link Author and Book objects, like the usage of @Entity, @Id, @OneToMany, ... like described here https://www.baeldung.com/spring-hibernate-n1-problem 
sorry, but I don't understand how it works without JPA annotations ...